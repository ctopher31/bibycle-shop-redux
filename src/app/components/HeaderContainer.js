import { connect } from 'react-redux';
import Header from './Header';

const mapStateToProps = state => ({ cartCount: state.cart.cartCount });

export default connect(
  mapStateToProps
)(Header);
