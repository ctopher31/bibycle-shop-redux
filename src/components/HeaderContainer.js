import { connect } from 'react-redux';
import Header from './Header';

const mapStateToProps = state => ({ cartCount: state.cartCount, });

export default connect(mapStateToProps, null)(Header);
