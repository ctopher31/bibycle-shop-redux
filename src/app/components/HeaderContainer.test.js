import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { mount } from 'enzyme';
import cartReducer from '../../cart/reducers';
import HeaderContainer from './HeaderContainer';
import Header from './Header';

describe('HeaderContainer', () => {
  it('renders the header', () => {
    const initialState = {
      cart: {
        cartCount: 1,
      },
    };

    const store = createStore(cartReducer, initialState);
    const wrapper = mount(
      <Router>
        <Provider store={store}>
          <HeaderContainer />
        </Provider>
      </Router>
    );

    expect(wrapper.find(HeaderContainer).length).toEqual(1);
    const container = wrapper.find(HeaderContainer);
    expect(container.find(Header).length).toEqual(1);
    expect(store.getState().content).toMatchSnapshot();
  });
});
