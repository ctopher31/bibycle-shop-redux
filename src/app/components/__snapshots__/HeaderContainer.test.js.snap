// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer renders the header 1`] = `undefined`;
